### PR TITLE
Rom tests

### PR DESCRIPTION
**Pull request by vanakenm**

### CODE REVIEW

File: src/FileExplorer.js [hunk 1]
```diff
@@ -104,11 +106,15 @@ function FileExplorer(props) {
 
         function open()
         {
-            // its important to set the current path first !!
-            getFilesFromResourceURL(url).then((fileArray) => {
-                setCurrentPath(url);
-                setFiles(fileArray);
-            });
+            if (url[url.length - 1] === "/") {
```
Comment (vanakenm) [inline src/FileExplorer.js]: .endsWith("/") exist in JS - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith - does the same but generally more readable

File: src/FileUpload.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+function FileUpload(props)
+{
+	let currentPath = props.currentPath;
+	let [selectedFiles, setSelectedFiles] = useState([]);
+
+
+	function upload() 
+	{
+	    if(selectedFiles.length != 0)
```
Comment (vanakenm) [inline src/FileUpload.js]: Maybe remove the console.logs if we don't need them anymore?

File: src/FileUpload.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	    console.log("uploading files ...");
+	    for (let file of selectedFiles)
+	    {
+	      	placeFileInContainer(file, currentPath);
+	    }
+	}
+
+
+	/**
+	 * Upload file into the targetContainer.
```
Comment (vanakenm) [inline src/FileUpload.js]: I'm not against comments but not sure this one helps

File: src/FileUpload.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	 * @param  {[type]} file               [description]
+	 * @param  {[type]} targetContainerURL [description]
+	 * @return {[type]}                    [description]
+	 */
+	async function placeFileInContainer(file, targetContainerURL) {
+	  try 
+	  {
+  		const savedFile = await saveFileInContainer(
```
Comment (vanakenm) [inline src/FileUpload.js]: Indent/format properly (VS code can do that for you)

File: src/FileUpload.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+	{
+		document.querySelector("#file-input").click();
+	}
+
+
+	function selectedFilesToReact()
+	{
+		let res = [];
+
+		if (selectedFiles.length > 0)
```
Comment (vanakenm) [inline src/FileUpload.js]: You don't need this - if the length is at 0, the for will not run at all - same result

File: src/FileUpload.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+		{
+			for (let file of selectedFiles)
```
Comment (vanakenm) [inline src/FileUpload.js]: JavaScript has a "map" function - something to transform an array into another array

res = selectedFiles.map(sf => sf.name)

**Review by vanakenm — COMMENTED** [review]
Nice job - left some comments to help cleanup / propose some alternatives

File: src/FileUpload.js (hunk 3 above)
Comment (barbarebelge) [reply src/FileUpload.js]: it's a reflex :) I just add default javadoc/doxygen comments to the functions generally. But yeah in this case this doesn't help much, we will probably have to improve it

File: src/FileExplorer.js (hunk 1 above)
Comment (barbarebelge) [reply src/FileExplorer.js]: very good idea !

File: src/FileUpload.js (hunk 2 above)
Comment (Speedbuilder) [reply src/FileUpload.js]: done

File: src/FileUpload.js (hunk 4 above)
Comment (Speedbuilder) [reply src/FileUpload.js]: done

File: src/FileUpload.js (hunk 6 above)
Comment (Speedbuilder) [reply src/FileUpload.js]: this cannot be resolved, selectedFiles is iterable but not of type array, so map() and forEach() functions don't work on this object

**Pull request merged**